lib/util/sdp/index.js

```diff
@@ -31,23 +41,28 @@ function createBLine(modifier, maxBitrate) {
 /**
  * Create a Codec Map for the given m= section.
  * @param {string} section - The given m= section
- * @returns {Map<AudioCodec|VideoCodec, Array<PayloadType>>}
+ * @returns {Map<Codec, Array<PT>>}
  */
 function createCodecMapForMediaSection(section) {
-  var payloadTypes = getPayloadTypesInMediaSection(section);
-  return payloadTypes.reduce(function(codecMap, payloadType) {
-    var rtpmapPattern = new RegExp('a=rtpmap:' + payloadType + ' ([^/]+)');
-    var matches = section.match(rtpmapPattern);
-
-    // This should not happen, but in case there is no rtpmap entry
-    // for a given Codec Name, then return the existing Codec Map.
-    if (!matches) {
-      return codecMap;
-    }
+  return Array.from(createPtToCodecName(section)).reduce(function(codecMap, pair) {
+    var pt = pair[0];
+    var codecName = pair[1];
+    var pts = codecMap.get(codecName) || [];
+    return codecMap.set(codecName, pts.concat(pt));
+  }, new Map());
+}
 
-    var codecName = matches[1].toLowerCase();
-    var types = codecMap.get(codecName) || [];
-    return codecMap.set(codecName, types.concat(payloadType));
+/**
+ * Create a Map from PTs to codec names for the given m= section.
+ * @param {string} mediaSection - The given m= section.
+ * @returns {Map<PT, Codec>} ptToCodecName
+ */
+function createPtToCodecName(mediaSection) {
+  return getPayloadTypesInMediaSection(mediaSection).reduce(function(ptToCodecName, pt) {
+    var rtpmapPattern = new RegExp('a=rtpmap:' + pt + ' ([^/]+)');
+    var matches = mediaSection.match(rtpmapPattern);
+    var codecName = matches ? matches[1].toLowerCase() : '';
```

Comment:
@markandrus Why are we adding an empty string for the payload type whose codec name doesn't exist?

Reply:
Only dynamic PTs are required to have "a=rtpmap" entries—there are some that do not require these attributes. I guess we could/should look up their names statically, though? We don't really need/use them, though. The Paper doc where I describe this mentions letting the `codecName` be the empty string, though.